crates/shared/src/price_estimation/native_price_cache.rs

```diff
@@ -106,42 +108,39 @@ impl Inner {
     /// have fetched some requested price in the meantime.
     fn estimate_prices_and_update_cache<'a>(
         &'a self,
-        tokens: &'a [H160],
+        tokens: impl Iterator<Item = &'a H160> + Send + 'a,
         max_age: Duration,
         parallelism: usize,
     ) -> futures::stream::BoxStream<'_, (usize, NativePriceEstimateResult)> {
-        let estimates = tokens
-            .iter()
-            .enumerate()
-            .map(move |(index, token)| async move {
-                {
-                    // check if price is cached by now
-                    let now = Instant::now();
-                    let mut cache = self.cache.lock().unwrap();
-                    let price = Self::get_cached_price(*token, now, &mut cache, &max_age, false);
-                    if let Some(price) = price {
-                        return (index, price);
-                    }
+        let estimates = tokens.enumerate().map(move |(index, token)| async move {
+            {
```

Comment:
Only indented 1 level to the left because the `.map()` now fits into 1 line due to using `impl Iterator` now.